sdk/core/core-lro/review/core-lro.api.md

```diff
@@ -9,6 +9,65 @@ import { AbortSignalLike } from '@azure/abort-controller';
 // @public
 export type CancelOnProgress = () => void;
 
+// @public
+export function createGetLroStatusFromResponse<TResult>(lroPrimitives: LongRunningOperation<TResult>, config: LroConfig, lroResourceLocationConfig?: LroResourceLocationConfig): GetLroStatusFromResponse<TResult>;
+
+// @public
+export type GetLroStatusFromResponse<T> = (rawResponse: RawResponse, flatResponse: T) => LroStatus<T>;
+
+// @public
+export interface LongRunningOperation<T> {
+    requestMethod: string;
+    requestPath: string;
+    retrieveAzureAsyncResource: (path?: string) => Promise<LroResponse<T>>;
+    sendInitialRequest: (initializeState: (rawResponse: RawResponse, flatResponse: unknown) => boolean) => Promise<LroResponse<T>>;
+    sendPollRequest: (config: LroConfig, path: string) => Promise<LroStatus<T>>;
+}
+
+// @public
+export interface LroConfig {
+    mode?: LroMode;
+    resourceLocation?: string;
+}
+
+// @public
+export class LroEngine<TResult, TState extends PollOperationState<TResult>> extends Poller<TState, TResult> {
+    constructor(lro: LongRunningOperation<TResult>, options?: LroEngineOptions);
+    delay(): Promise<void>;
+    }
+
+// @public
+export interface LroEngineOptions {
+    intervalInMs?: number;
+    resumeFrom?: string;
+}
+
+// @public
+export interface LroInProgressState<T> extends LroResponse<T> {
+    done: false;
+    next?: () => Promise<LroStatus<T>>;
+}
+
+// @public
+export type LroMode = "AzureAsync" | "Location" | "Body";
+
+// @public
+export type LroResourceLocationConfig = "azure-async-operation" | "location" | "original-uri";
```

Comment:
Would extensible enums work here? (also on LroMode)?

Reply:
I do not think so because this is a defined swagger option. `LroMode` is no longer part of the public surface.